### PR TITLE
Report updates

### PR DESCRIPTION
**Pull request by fasterius**
This PR contains substantial changes and fixes to the downstream analysis reports. Changes include:

- Changing the documentation from scanpy to custom explanations and contexts
- Harmonisation of input/output file/process/parameter names
- Renaming of variables/parameters to better describe what they are used for and to adhere to PEP8
- Merge the pre/post-processing subworkflows into one `downstream` subworkflow
- Moving the normalization from the QC report to the clustering report
- Store raw data in the AnnData object so that a user may re-do analyses from scratch if so desired
- Simplified input file staging in reports
- Addition of filtering of mitochondrial, ribosomal and haemoglobin content
- Additional figures
- Removal of redundant figures
- Optimisation of figure sizes to fit the report format, where possible
- Minor formatting and language changes throughout
- Updated documentation

Hopefully this PR makes the reports and their related pipeline content are a bit more fully-featured and better documented.

## PR checklist

- [X] This comment contains a description of changes (with reason).
- [X] If you've fixed a bug or added code that should be tested, add tests!
- [X] Make sure your code lints (`nf-core lint`).
- [X] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [X] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [X] Usage Documentation in `docs/usage.md` is updated.
- [X] Output Documentation in `docs/output.md` is updated.
- [X] `CHANGELOG.md` is updated.

### CODE REVIEW

File: bin/st_quality_controls.qmd [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+
+```{python}
+#| layout-nrow: 2
+sc.pl.spatial(st_adata, color = ["total_counts", "n_genes_by_counts"], size=1.25)
+sc.pl.spatial(st_adata, color = ["pct_counts_mt", "pct_counts_ribo", "pct_counts_hb"], size=1.25)
+```
+
+## Scatter plots
+
+It is also useful to some of these quality metrics against each other in scatter
```
Comment (cavenel) [inline bin/st_quality_controls.qmd]: also useful to **compare**?

File: nextflow_schema.json [hunk 2]
```diff
@@ -95,41 +95,48 @@
...
                 },
-                "st_preprocess_min_cells": {
+                "st_qc_min_spots": {
                     "type": "integer",
                     "default": 1,
                     "description": "The minimum number of spots in which a gene is expressed for that gene to pass the filtering.",
                     "fa_icon": "fas fa-hashtag"
                 },
-                "st_preprocess_hist_qc_max_total_counts": {
+                "st_qc_mito_threshold": {
```
Comment (cavenel) [inline nextflow_schema.json:116]: Any reason to have integer instead of float here?

**Review by cavenel — COMMENTED** [review]
Some minor comments and then all good for me!

File: bin/st_quality_controls.qmd (hunk 1 above)
Comment (fasterius) [reply bin/st_quality_controls.qmd]: Thanks, I missed that!

File: nextflow_schema.json (hunk 2 above)
Comment (fasterius) [reply nextflow_schema.json:116]: Nope! I changed it now.

**Review by cavenel — APPROVED** [review]
All good!

**Pull request merged**